plugin/storage/integration/es_index_cleaner_test.go

```diff
@@ -123,6 +125,87 @@ func TestIndexCleaner(t *testing.T) {
 	}
 }
 
+func TestIndexRollover_FailIfILMNotPresent(t *testing.T) {
+	client, err := createESClient()
+	require.NoError(t, err)
+	// make sure ES is clean
+	cleanES(t, client, "jaeger-ilm-policy")
+	envVars := []string{"ES_USE_ILM=true"}
+	errorMsg := "exit status 1"
+	er := runEsRollover("init", envVars)
+	assert.Equal(t, errorMsg, er.Error())
+	indices, err1 := client.IndexNames()
+	require.NoError(t, err1)
+	assert.Empty(t, indices)
+}
+
+func TestIndexRollover_CreateIndicesWithILM(t *testing.T) {
+	client, err := createESClient()
+	require.NoError(t, err)
+
+	esVersion, ev := getVersion(client)
+	require.NoError(t, ev)
+
+	if esVersion != 7 {
+		er := runEsRollover("init", []string{"ES_USE_ILM=true"})
+		assert.Equal(t, "exit status 1", er.Error())
+		indices, err1 := client.IndexNames()
+		require.NoError(t, err1)
+		assert.Empty(t, indices)
+
+	} else {
+		envVars := []string{"ES_USE_ILM=true"}
+		expectedIndices := []string{"jaeger-span-000001", "jaeger-service-000001"}
+		t.Run(fmt.Sprintf("%s_no_prefix", "CreateIndicesWithILM"), func(t *testing.T) {
+			runIndexRolloverWithILMTest(t, client, "", expectedIndices, envVars)
+		})
+		t.Run(fmt.Sprintf("%s_prefix", "CreateIndicesWithILM"), func(t *testing.T) {
+			runIndexRolloverWithILMTest(t, client, indexPrefix, expectedIndices, append(envVars, "INDEX_PREFIX="+indexPrefix))
+		})
+	}
+}
+
+func runIndexRolloverWithILMTest(t *testing.T, client *elastic.Client, prefix string, expectedIndices, envVars []string) {
+	writeAliases := []string{"jaeger-service-write", "jaeger-span-write"}
+
+	// make sure ES is clean
+	cleanES(t, client, "jaeger-ilm-policy")
+	erILM := createILMPolicy(client, "jaeger-ilm-policy")
+	require.NoError(t, erILM)
+
+	if prefix != "" {
+		prefix = prefix + "-"
+	}
+	var expected, expectedWriteAliases, actualWriteAliases []string
+	for _, index := range expectedIndices {
+		expected = append(expected, prefix+index)
+	}
+	for _, alias := range writeAliases {
+		expectedWriteAliases = append(expectedWriteAliases, prefix+alias)
+	}
+
+	//run rollover with given EnvVars
+	e1 := runEsRollover("init", envVars)
+	require.NoError(t, e1)
+
+	indices, e2 := client.IndexNames()
+	require.NoError(t, e2)
+
+	//Get ILM Policy Attached
```

Comment:
```suggestion
	// Get attached ILM Policy
```

Reply:
nit: It would be more readable if this is grouped with the `// Check ILM Policy is attached` code block.

Reply:
Agreed and moved